### ISSUE TITLE
add development dependencies to project.json

### ISSUE THREAD

**Issue by enricosada** [dotnet/cli#646]
I think it should be awesome to have some place to define the **build dependencies** ( nuget packages with coreclr target ) of the project instead of only the package dependencies.

for example, i have a tool `HelperStuff` program who run on coreclr.

i can add this on package on a 

```json
    "build-dependencies": {
        "HelperStuff": "1.0.0"
    },
```

and i can run this program  on postbuild, prebuild event.

Each package can use his version of the `HelperStuff`, no need to globally use a specific version.

It's usefull for 

- create custom documentation
- generate or update some files
- extend where possibile

This work it's done on the `dotnet restore` step.

This can be a shared option in global.json of repo

**Comment by davidfowl** [comment]
See https://github.com/piotrpMSFT/cli/blob/PackagedCommands/test/PackagedCommands/Consumers/AppWithToolDependency/project.json

**Comment by enricosada** [comment]
really nice, thx @davidfowl . 
I tried to find docs and search code before opening this issue, but the code change fast :smile:

**Comment by enricosada** [comment]
i'll leave it open, is a work in progress atm, so if someone search about it, can see the work of @piotrpMSFT

**Comment by piotrpMSFT** [comment]
:) thanks!

**Comment by blackdwarf** [comment]
Fixed in #697 so closing.
